Add kubestash docs (#1)
kubestash/docs: docs/concepts/crds/backupsession/index.md

section_menu_id: concepts
---

> New to KubeStash? Please start [here](/docs/concepts/README.md).

# BackupSession

## What is BackupSession

> A `BackupSession` is a Kubernetes `CustomResourceDefinition`(CRD) which represents a backup run of the respective target referenced by a `BackupConfiguration` in a Kubernetes native way.
> 

Reviewer: ```suggestion
A `BackupSession` is a Kubernetes `CustomResourceDefinition`(CRD) which represents a backup run triggered for a session of a `BackupConfiguration` for a target application.

```

#### status.nextRetry

`status.nextRetry` specifies the timestamp when this backup will be retried if it has failed. This field will exist only if the `retryConfig` has been set in the respective backup invoker.

#### status.conditions

`status.conditions` shows the conditions of different operations/steps of the backup process. The following conditions are set by the KubeStash operator on a `BackupSession`.

| Condition Type                      | Usage                                                                                      |
|-------------------------------------|--------------------------------------------------------------------------------------------|
| `BackupSkipped`                     | indicates that the current session was skipped.                                            |
| `SessionHistoryCleaned`             | indicates whether the backup history was cleaned or not according to `backupHistoryLimit`. |
| `PreBackupHooksExecutionSucceeded`  | indicates whether the pre-backup hooks were executed successfully or not.                  |

> | `PostBackupHooksExecutionSucceeded` | indicates whether the pre-backup hooks were executed successfully or not.                  |

Reviewer: ```suggestion
| `PostBackupHooksExecutionSucceeded` | indicates whether the post-backup hooks were executed successfully or not.                  |
```

> | `BackupExecutorEnsured`             | indicates whether the Backup Executor is ensured or not.                                   |
> | `SnapshotsEnsured`                  | indicates whether Snapshots are ensured for each Repository or not                         |
> | `DeadlineExceeded`                  | indicates whether the session deadline was exceeded or not.                                |
> | `MetricsPushed`                     | indicates whether Metrics are pushed or not                                                |

Reviewer: ```suggestion
| `BackupExecutorEnsured`             | indicates whether the Backup Executor was ensured or not.                                   |
| `SnapshotsEnsured`                  | indicates whether Snapshots were ensured for each Repository or not                         |
| `DeadlineExceeded`                  | indicates whether the session deadline was exceeded or not.                                |
| `MetricsPushed`                     | indicates whether Metrics were pushed or not                                                |
```
